### ISSUE TITLE
Move goal postprocessing in fuse-ignite-install's release.sh to templates

### ISSUE THREAD

**Issue by paoloantinori** [syndesisio/syndesis#2639]

**Comment by rhuss** [comment]
@paoloantinori what is this issue about and what is the motivation ? Or stated otherwise: What is not working ?

**Comment by paoloantinori** [comment]
We have part of the logic that discriminate based on the target environment directly in the OpenShift template.
Another part of logic is instead performed directly in https://github.com/syndesisio/fuse-ignite-install repo.

We should converge to a single place to keep track of all the configuration management related chaning

**Comment by rhuss** [comment]
Could you leverage where in syndesisio/fuse-ignite-install we are discriminating on the target platform ? There is no upstream part in fuse-ignite-install.

The ideal solution would be:

* Have the template templates in an extra repo along with the generator, let's say "syndesiso/template-generator"
* Have "syndesisio/syndesis" use the generator and template templates from this repository to generate the upstream templates.
* Have "syndesis/fuse-ignite-install" use the generator and template templates from this repository to generate the fuset-ignite templates.

For simplicity reason this "template-generator" is kept in our mono repo. We could avoid the conditionals in the templates in syndesis only if we would *copy over and duplicate* the functionality, including the template templates. Maintaining both of them and keeping them synced shouldn't be the target imo.

long story short: I'm pretty happy with the current setup.

**Comment by rhuss** [comment]
btw, should we rename fuse-ignite-install to fuse-online-install in the near future ?

**Comment by paoloantinori** [comment]
> Could you leverage where in syndesisio/fuse-ignite-install we are discriminating on the target platform ?

- https://github.com/syndesisio/fuse-ignite-install/blob/master/release.sh#L207-L209
- https://github.com/syndesisio/fuse-ignite-install/blob/master/release.sh#L214-L216


probably other places.

Considering in the templates we currently have optional sections related to the environment, all this logic should live in a single repository, not in both, to simplify the code.


And yes, since we have dropped Ignite, probably it's better to rename the repo.

**Comment by paoloantinori** [comment]
> I hope you agree, that these are really hacks which we have to remove. 

agree this is the bulk of the activity. to push *every* platform dependent  logic in a single place, either on the main repository or on the install one.

**Comment by heiko-braun** [comment]
Related to #2687
